test/integ/pandas/test_arrow_pandas.py

```diff
@@ -567,6 +569,118 @@ def test_timestampltz(conn_cnx, scale, timezone):
     finish(conn_cnx, table)
 
 
+def do_validate_pandas(
+    cnx_table,
+    sql,
+    cases,
+    col_count,
+    method="one",
+    data_type="float",
+    epsilon=None,
+    scale=0,
+    timezone=None,
+    telemetry_test=None,
+    telemetry_to_check=None,
+):
+    row_count = len(cases)
+    assert col_count != 0, "# of columns should be larger than 0"
+
+    cursor_table = cnx_table.cursor()
+    cursor_table.execute(SQL_ENABLE_ARROW)
+    cursor_table.execute(sql)
+
+    # build dataframe
+    total_rows, total_batches = 0, 0
+    start_time = time.time()
+
+    if method == "one":
+        df_new = cursor_table.fetch_pandas_all()
+        total_rows = df_new.shape[0]
+    else:
+        for df_new in cursor_table.fetch_pandas_batches():
+            total_rows += df_new.shape[0]
+            total_batches += 1
+
+    if telemetry_to_check is not None and telemetry_test is not None:
+        assert any(
+            t.message["type"] == telemetry_to_check for t in telemetry_test.records
+        )
```

Comment:
I think this could live inside of the tests and then we wouldn't necessarily need the helper function refactor. Other than this change this helper function looks the same as `validate_pandas`.

Reply:
I think this issue is one cons in using fixture. connection object was initialized not directly in test_** so the conn_cnx has to be passed into validate_pandas function where we get the actual connection object out of the fixture. Then telemetry can be initialized upon this connection. But if we want to use the telemetry object some where else far away from where we initialize these fixtures, we has to pass around. Like this one, we could do the assert outside in test, but the telemetry object initialized in validate_pandas also has to return to the caller of validate_pandas.

Reply:
I don't see how that's the fixture's fault, it sounds like to me that it's the helper function's fault. We could have the helper accept `SnowflakeConnection` instead of the `Callable[..., "SnowflakeConnection"]` that `conn_cnx` is and then use that connection in the helper function.

Reply:
Providing SnowflakeConnection to helper functions, e.g. init, finish, validate_pandas means unnecessarily rewrite all tests that call these these functions and put these calls under "with...". The reason to use help functions like what I am doing right now is to maintain the same validate_pandas for other supposedly untouched tests to use, then only the newly added test needs to look at telemetry data. So the added complexity is contained only in new code. I feel this is simpler. What do you think?